### PR TITLE
fix hrm error when works with Module federation

### PR DESCRIPTION
**Pull request by wangjinyang**
**What kind of change does this PR introduce?**
fixes [#15534](https://github.com/webpack/webpack/issues/15534)

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
no

### CODE REVIEW

File: lib/container/RemoteRuntimeModule.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ class RemoteRuntimeModule extends RuntimeModule {
 							Template.indent(
 								`error.message += '\\nwhile loading "' + data[1] + '" from ' + data[2];`
 							),
-							`__webpack_modules__[id] = ${runtimeTemplate.basicFunction("", [
+							`__webpack_require__.m[id] = ${runtimeTemplate.basicFunction("", [
```
Comment (sokra) [inline lib/container/RemoteRuntimeModule.js:76]:
```suggestion
							`${RuntimeGlobals.moduleFactories}[id] = ${runtimeTemplate.basicFunction("", [
```

File: lib/container/RemoteRuntimeModule.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ class RemoteRuntimeModule extends RuntimeModule {
 						)};`,
 						`var onFactory = ${runtimeTemplate.basicFunction("factory", [
 							"data.p = 1;",
-							`__webpack_modules__[id] = ${runtimeTemplate.basicFunction(
+							`__webpack_require__.m[id] = ${runtimeTemplate.basicFunction(
```
Comment (sokra) [inline lib/container/RemoteRuntimeModule.js:114]:
```suggestion
							`${RuntimeGlobals.moduleFactories}[id] = ${runtimeTemplate.basicFunction(
```

**Review by sokra — CHANGES_REQUESTED** [review]
That looks good. Only a nitpick is to get the name of the property from the constants.

File: lib/container/RemoteRuntimeModule.js (hunk 2 above)
Comment (wangjinyang) [reply lib/container/RemoteRuntimeModule.js]: many thanks for tips

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**